packages/google_maps_flutter/lib/src/location.dart

```diff
@@ -18,7 +18,8 @@ class LatLng {
         assert(longitude != null),
         latitude =
             (latitude < -90.0 ? -90.0 : (90.0 < latitude ? 90.0 : latitude)),
-        longitude = (longitude + 180.0) % 360.0 - 180.0;
+        longitude =
+            longitude > 180 ? (longitude + 180.0) % 360.0 - 180.0 : longitude;
```

Comment:
I think this should be

`longitude = (longitude < -180.0 || longitude >= 180.0) ? (longitude + 180.0) % 360.0 - 180.0 : longitude;`

For two reasons:
- As it stands the new code will map +180 to +180, whereas the old code mapped it to 0. The convention for longitude is normally that it should be in the interval [-180, +180), and that interval doesn't include +180.
- The new code doesn't wrap values less than -180 at all. The old code did.

Reply:
You're right